src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx

```diff
@@ -0,0 +1,136 @@
+import type {ForwardedRef} from 'react';
+import React, {forwardRef, useCallback} from 'react';
+import {View} from 'react-native';
+// eslint-disable-next-line no-restricted-imports
+import type {ScrollView as ScrollViewRN} from 'react-native';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type {ApprovalWorkflowOnyx} from '@src/types/onyx';
+import type {Approver} from '@src/types/onyx/ApprovalWorkflow';
+
+type ApprovalWorkflowEditorProps = {
+    approvalWorkflow: ApprovalWorkflowOnyx;
+    policyID: string;
+};
+
+function ApprovalWorkflowEditor({approvalWorkflow, policyID}: ApprovalWorkflowEditorProps, ref: ForwardedRef<ScrollViewRN>) {
+    const styles = useThemeStyles();
+    const {translate, toLocaleOrdinal} = useLocalize();
+
+    const approverDescription = useCallback(
+        (index: number) =>
+            approvalWorkflow.approvers.length > 1 ? `${toLocaleOrdinal(index + 1, true)} ${translate('workflowsPage.approver').toLowerCase()}` : `${translate('workflowsPage.approver')}`,
+        [approvalWorkflow.approvers.length, toLocaleOrdinal, translate],
+    );
+    const members = approvalWorkflow.isDefault ? translate('workspace.common.everyone') : approvalWorkflow.members.map((m) => m.displayName).join(', ');
+
+    const approverErrorMessage = useCallback(
+        (approver: Approver | undefined, approverIndex: number) => {
+            const previousApprover = approvalWorkflow.approvers.slice(0, approverIndex).filter(Boolean).at(-1);
+            const error = approvalWorkflow?.errors?.[`approver-${approverIndex}`];
+
+            if (!error) {
+                return;
+            }
+
+            if (error === 'workflowsPage.approverCircularReference') {
+                if (!previousApprover || !approver) {
+                    return;
+                }
+                return translate('workflowsPage.approverCircularReference', {
+                    name1: approver?.displayName,
+                    name2: previousApprover.displayName,
+                });
+            }
+
+            return translate(error);
+        },
+        [approvalWorkflow.approvers, approvalWorkflow.errors, translate],
+    );
+
+    const approverHintMessage = useCallback(
+        (approver: Approver | undefined, approverIndex: number) => {
+            const previousApprover = approvalWorkflow.approvers.slice(0, approverIndex).filter(Boolean).at(-1);
+            if (approver?.isInMultipleWorkflows && !!previousApprover) {
```

Comment:
@blazejkustra, I am not sure why we use this condition.

Let's see this case:

<img width="1051" alt="Screenshot 2024-08-12 at 18 02 24" src="https://github.com/user-attachments/assets/8fbf831c-b562-49c6-8cf6-f04b82a83da3">

`csciolla1@gmail.com` belongs to two different approval workflows. There are no circular reference 

The hint message: "csciolla1@gmail.com already approves reports to m in a separate workflow" is wrong

Reply:
Wdyt if we add one more condition:

```
previousApprover.forwardsTo === approver.email

```

Reply:
This is brilliant @DylanDylann, this made the logic so much easier. Thank you for this idea! Here the commit: https://github.com/Expensify/App/pull/46798/commits/f554da6696005e996aa1f7aa729d9519b763d108

Reply:
@DylanDylann I landed on this logic, thanks for your suggestion: https://github.com/Expensify/App/pull/46798/commits/d74ed36405e45e8010e517b8c2f8d94e88d321a3

Reply:
@blazejkustra Are you planning to create the PR for [this issue](https://github.com/Expensify/App/issues/45953)? If so, please tag me on that PR even if it's just a draft PR. Thus, I can get an overview of this feature. Rest assured, I will remain focused on the current PR. Thank you.

Reply:
There is a draft for it already: https://github.com/Expensify/App/pull/47249/files